diff -r 409691084d19 src/os_unix.c
--- a/src/os_unix.c	Tue Oct 04 21:22:44 2011 +0200
+++ b/src/os_unix.c	Thu Oct 06 22:01:17 2011 -0400
@@ -5146,11 +5146,21 @@
 # endif
 # ifdef EINTR
 	if (ret == -1 && errno == EINTR)
+	{
+
+	    /*
+	     * Check whether window has been resized (EINTR is due to SIGWINCH)
+	     */
+	    if (do_resize)
+		handle_resize();
+
 	    /* Interrupted by a signal, need to try again.  We ignore msec
 	     * here, because we do want to check even after a timeout if
 	     * characters are available.  Needed for reading output of an
 	     * external command after the process has finished. */
 	    goto select_eintr;
+
+	}
 # endif
 # ifdef __TANDEM
 	if (ret == -1 && errno == ENOTSUP)
